plugins/modules/win_scoop_bucket.ps1

```diff
@@ -0,0 +1,166 @@
+#!powershell
+
+# Copyright: (c) 2020, Jamie Magee <jamie.magee@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#Requires -Module Ansible.ModuleUtils.ArgvParser
+#Requires -Module Ansible.ModuleUtils.CommandUtil
+#AnsibleRequires -CSharpUtil Ansible.Basic
+
+$spec = @{
+  options             = @{
+    name  = @{ type = "str"; required = $true }
+    repo  = @{ type = "str" }
+    state = @{ type = "str"; default = "present"; choices = "present", "absent" }
+  }
+  supports_check_mode = $true
+}
+
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
+
+$name = $module.Params.name
+$repo = $module.Params.repo
+$state = $module.Params.state
+
+function Install-Scoop {
```

Comment:
I would be against having `win_scoop_bucket` actually install Scoop. For `win_scoop` that's fine as it matches the behaviour of `win_chocolatey` for but extra functionality like buckets it's best to just have it a requirement that Scoop is already installed.

This way you save on duplicating code, simplify the module, and keep the module focused on one aspect and doing that well.